include/alpaka/core/ApiCudaRt.hpp

```diff
@@ -0,0 +1,357 @@
+        static constexpr auto deviceAttributeMaxGridDimX = ::cudaDevAttrMaxGridDimX;
+        static constexpr auto deviceAttributeMaxGridDimY = ::cudaDevAttrMaxGridDimY;
+        static constexpr auto deviceAttributeMaxGridDimZ = ::cudaDevAttrMaxGridDimZ;
+        static constexpr auto deviceAttributeMaxSharedMemoryPerBlock = ::cudaDevAttrMaxSharedMemoryPerBlock;
+        static constexpr auto deviceAttributeMaxThreadsPerBlock = ::cudaDevAttrMaxThreadsPerBlock;
+        static constexpr auto deviceAttributeMultiprocessorCount = ::cudaDevAttrMultiProcessorCount;
+
+        static constexpr auto limitPrintfFifoSize = ::cudaLimitPrintfFifoSize;
+
+        // runtime API
+        static inline cudaError_t deviceGetAttribute(int* value, enum cudaDeviceAttr attr, int device)
+        {
+            return ::cudaDeviceGetAttribute(value, attr, device);
+        }
+
+        static inline cudaError_t deviceGetLimit(size_t* pValue, enum cudaLimit limit)
+        {
+            return ::cudaDeviceGetLimit(pValue, limit);
+        }
+
+        static inline cudaError_t deviceReset(void)
+        {
+            return ::cudaDeviceReset();
+        }
+
+        static inline cudaError_t deviceSetLimit(enum cudaLimit limit, size_t value)
+        {
+            return ::cudaDeviceSetLimit(limit, value);
+        }
+
+        static inline cudaError_t deviceSynchronize(void)
+        {
+            return ::cudaDeviceSynchronize();
+        }
+
+        static inline cudaError_t eventCreate(cudaEvent_t* event)
+        {
+            return ::cudaEventCreate(event);
+        }
+
+        static inline cudaError_t eventCreateWithFlags(cudaEvent_t* event, unsigned int flags)
+        {
+            return ::cudaEventCreateWithFlags(event, flags);
+        }
+
+        static inline cudaError_t eventDestroy(cudaEvent_t event)
+        {
+            return ::cudaEventDestroy(event);
+        }
+
+        static inline cudaError_t eventQuery(cudaEvent_t event)
+        {
+            return ::cudaEventQuery(event);
+        }
+
+        static inline cudaError_t eventRecord(cudaEvent_t event, cudaStream_t stream)
+        {
+            return ::cudaEventRecord(event, stream);
+        }
+
+        static inline cudaError_t eventSynchronize(cudaEvent_t event)
+        {
+            return ::cudaEventSynchronize(event);
+        }
+
+        static inline cudaError_t free(void* devPtr)
+        {
+            return ::cudaFree(devPtr);
+        }
+
+        static inline cudaError_t freeAsync(void* devPtr, cudaStream_t hStream)
+        {
+#    if BOOST_LANG_CUDA >= BOOST_VERSION_NUMBER(11, 2, 0)
+            return ::cudaFreeAsync(devPtr, hStream);
+#    else
+            // Unused.
+            static_cast<void>(devPtr);
+            static_cast<void>(hStream);
+            // Not implemented.
+            return ::cudaErrorUnknown;
+#    endif
+        }
+
+        static inline cudaError_t funcGetAttributes(struct cudaFuncAttributes* attr, const void* func)
```

Comment:
```suggestion
        static inline cudaError_t funcGetAttributes(cudaFuncAttributes* attr, const void* func)
```
We don't need the elaborated type specifier here. Applies to other places as well.

Reply:
Right... I copied the signatures from the CUDA header, but you are right they can be simplified for the C++-only case (like Alpaka).